src/track/beats.h

```diff
@@ -54,124 +84,264 @@ class Beats : public QObject {
         THREEHALVES,
     };
 
-    virtual Beats::CapabilitiesFlags getCapabilities() const = 0;
-
-    // Serialization
-    virtual QByteArray toByteArray() const = 0;
-    virtual BeatsPointer clone() const = 0;
-
-    // A string representing the version of the beat-processing code that
-    // produced this Beats instance. Used by BeatsFactory for associating a
-    // given serialization with the version that produced it.
-    virtual QString getVersion() const = 0;
-    // A sub-version can be used to represent the preferences used to generate
-    // the beats object.
-    virtual QString getSubVersion() const = 0;
-
-    ////////////////////////////////////////////////////////////////////////////
-    // Beat calculations
-    ////////////////////////////////////////////////////////////////////////////
+    // TODO(JVC) Not needed
+    Beats::CapabilitiesFlags getCapabilities() const {
+        return BEATSCAP_TRANSLATE | BEATSCAP_SCALE | BEATSCAP_ADDREMOVE |
+                BEATSCAP_MOVEBEAT | BEATSCAP_HASBAR;
+    }
+    /// Serializes into a protobuff.
+    virtual QByteArray toProtobuff() const;
+    virtual BeatsPointer clone() const;
+
+    /// Returns a string representing the version of the beat-processing code that
+    /// produced this Beats instance. Used by BeatsFactory for associating a
+    /// given serialization with the version that produced it.
+    virtual QString getVersion() const;
```

Comment:
QString? Shouldn't this be an int?

Reply:
```cpp
    static const QString BEAT_MAP_VERSION;
    static const QString BEAT_GRID_1_VERSION;
    static const QString BEAT_GRID_2_VERSION;
```
These will eventually be removed so I won't change anything about them.
We will rely on the preference settings just to check whether the user wants constant or dynamic BPM. BeatsFactory will read the preference instead of this version.

Do we agree on entirely removing the Beats versioning?

Reply:
No, we cannot upgrade old data without versioning. The tricky situation we face now is that the legacy formats used a string to encode 2 different pieces of information, the version and the type (BeatMap or BeatGrid). Future upgrades will be more straightforward if we simply use an int to specify the version.